src/Nethermind/Nethermind.Init.Snapshot/InitDatabaseSnapshot.cs

```diff
@@ -116,9 +127,24 @@ private async Task InitDbFromSnapshot(CancellationToken cancellationToken)
             (await httpClient.SendAsync(request, HttpCompletionOption.ResponseHeadersRead, cancellationToken))
             .EnsureSuccessStatusCode();
 
+        FileMode snapshotFileMode = FileMode.Create;
+        if (snapshotFileInfo.Exists && response.StatusCode == HttpStatusCode.PartialContent)
+        {
+            snapshotFileMode = FileMode.Append;
+        }
+        else if (response.StatusCode == HttpStatusCode.OK)
```

Comment:
not sure about this and the next `else` given that we already [ensure a success status code a few lines above](https://github.com/NethermindEth/nethermind/pull/6309/files#diff-9d81e43b7817ba0d761c63003a8bfcba00705c6e4059cadd793e8cd32965f76eL117)

Reply:
`IsSuccessStatusCode => this._statusCode >= HttpStatusCode.OK && this._statusCode <= (HttpStatusCode) 299`
100 codes are success codes, we can understand only two